idx/idx.go

```diff
@@ -112,4 +112,7 @@ type MetricIndex interface {
 	Find(int, string, int64) ([]Node, error)
 	List(int) []Archive
 	Prune(int, time.Time) ([]Archive, error)
+	TagList(int, uint32) []string
+	Tag(int, string) map[string]uint32
+	IdsByTagExpressions(int, []string) ([]string, error)
```

Comment:
these 3 functions are not called by anything, it's not really clear what they are for

Reply:
adding comments